File: frontend/src/types/session.ts
export interface DeviceInfo {
  browser: string | null;
  browser_version: string | null;
  os: string | null;
  os_version: string | null;
  device_type: "desktop" | "mobile" | "tablet" | null;
}

export interface Session {
  id: string;
  device_info: DeviceInfo | null;
  ip_address: string | null;
  last_active: string | null;
  created_at: string;
  expires: string;
  is_current: boolean;
}

export interface SessionListResponse {
  sessions: Session[];
  total: number;
}

export interface RevokeSessionResponse {
  success: boolean;
  message: string;
  revoked_count: number;
}

export interface RevokeAllSessionsResponse {
  success: boolean;
  message: string;
  revoked_count: number;
}

File: backend/app/schemas/session.py
from typing import Optional
from pydantic import BaseModel, ConfigDict
import uuid
from datetime import datetime


class DeviceInfo(BaseModel):
    """Parsed device information."""
    browser: Optional[str] = None
    browser_version: Optional[str] = None
    os: Optional[str] = None
    os_version: Optional[str] = None
    device_type: Optional[str] = None  # desktop, mobile, tablet


class SessionSchema(BaseModel):
    """Schema for a user session."""
    id: uuid.UUID
    device_info: Optional[DeviceInfo] = None
    ip_address: Optional[str] = None
    last_active: Optional[datetime] = None
    created_at: datetime
    expires: datetime
    is_current: bool = False

    model_config = ConfigDict(from_attributes=True)


class SessionListResponse(BaseModel):
    """Response for listing sessions."""
    sessions: list[SessionSchema]
    total: int


class RevokeSessionResponse(BaseModel):
    """Response for revoking sessions."""
    success: bool
    message: str
    revoked_count: int = 1


class RevokeAllSessionsResponse(BaseModel):
    """Response for revoking all sessions."""
    success: bool
    message: str
    revoked_count: int
